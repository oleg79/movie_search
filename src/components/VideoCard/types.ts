export type VideoType = 'movie' | 'series' | 'episode';

export interface Video {
  Poster: string,
  Title: string,
  Type: VideoType,
  Year: string,
  imdbID: string,
}
