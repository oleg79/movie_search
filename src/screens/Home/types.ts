import {Video} from '../../components/VideoCard';

export interface SuccessResponse {
  Response: string;
  Search: Video[];
  totalResults: string;
}

export interface FailResponse {
  Response: string;
  Error: string;
}

export type ApiResponse = SuccessResponse | FailResponse | null;