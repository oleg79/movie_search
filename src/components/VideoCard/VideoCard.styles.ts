import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  card: {
    display: 'flex',
    flexDirection: 'row',
    width: '90%',
    alignSelf: 'center',
    marginVertical: 10,
    paddingVertical: 10,
  },

  poster: {
    width: 180,
    height: 200,
    resizeMode: 'contain',
  },

  infoContainer: {
    display: 'flex',
    justifyContent: 'space-evenly',
  },

  label: {
    fontSize: 12,
    color: '#37474f',
  },

  title: {
    flex: 1,
    flexWrap: 'wrap',
    width: 150
  },

  type: {
    color: '#fff',
    backgroundColor: '#e53935',
    borderRadius: 10,
    overflow: 'hidden',
    padding: 2,
    width: 60,
    textAlign: 'center',
    marginTop: 5,
  },
});
