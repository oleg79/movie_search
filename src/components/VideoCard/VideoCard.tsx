import * as React from 'react'
import {
  View,
  Text,
  Image
} from 'react-native';
import { Video } from './types';

// @ts-ignore
import NAImage from '../../images/not_available.png';

import styles from './VideoCard.styles';

interface Props {
  video: Video;
}

export default class VideoCard extends React.PureComponent<Props> {
  render() {
    const {
      Title,
      Year,
      Type,
      Poster,
    } = this.props.video;

    const imageSource = Poster === 'N/A' ? NAImage : { uri: Poster };

    return (
      <View>
        <Text>{ Title }</Text>
        <Image
          style={{
            width: 200,
            height: 200,
            resizeMode: 'contain'
          }}
          source={imageSource}
          defaultSource={NAImage}
        />
        <Text>{ Year }</Text>
        <Text>{ Type }</Text>
      </View>
    );
  }
}
