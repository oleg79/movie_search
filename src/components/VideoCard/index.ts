export { default as VideoCard } from './VideoCard'
export * from './types';
