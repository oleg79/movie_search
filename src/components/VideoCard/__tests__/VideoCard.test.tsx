import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import VideoCard from '../VideoCard';
import { Video } from '../types';

const video: Video = {
  Poster: "https://m.media-amazon.com/images/M/MV5BMjMxMTQ3MzMwMV5BMl5BanBnXkFtZTgwNTYxNzYxMTE@._V1_SX300.jpg",
  Title: "Cool Runnings",
  Type: "movie",
  Year: "1993",
  imdbID: "tt0106611",
};

describe('<VideoCard /> component', () => {

  it('Should be rendered properly', () => {
    const rendered = renderer.create(<VideoCard video={video}/>);
    const tree = rendered.toJSON();

    expect(tree).toMatchSnapshot();
  });

});
