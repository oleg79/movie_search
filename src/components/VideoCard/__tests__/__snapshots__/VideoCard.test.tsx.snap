// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VideoCard /> component Should be rendered properly 1`] = `
<View>
  <Text>
    Cool Runnings
  </Text>
  <Image
    defaultSource={
      Object {
        "testUri": "../../../src/images/not_available.png",
      }
    }
    source={
      Object {
        "uri": "https://m.media-amazon.com/images/M/MV5BMjMxMTQ3MzMwMV5BMl5BanBnXkFtZTgwNTYxNzYxMTE@._V1_SX300.jpg",
      }
    }
    style={
      Object {
        "height": 200,
        "resizeMode": "contain",
        "width": 200,
      }
    }
  />
  <Text>
    1993
  </Text>
  <Text>
    movie
  </Text>
</View>
`;
